Failing tests due to RecursionError

Two tests started failing when GAP is installed with a weird RecursionError. 

```python
>           assert np.allclose(config.coordinates, exp_vals['coords'][i])

tests/test_configuration_set.py:240: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../../micromamba/envs/gha-test-env/lib/python3.10/site-packages/numpy/_core/numeric.py:2329: in allclose
    res = all(isclose(a, b, rtol=rtol, atol=atol, equal_nan=equal_nan))
../../../micromamba/envs/gha-test-env/lib/python3.10/site-packages/numpy/_core/numeric.py:2449: in isclose
    | (x == y))
../../../micromamba/envs/gha-test-env/lib/python3.10/site-packages/autode/values.py:585: in __eq__
    and np.allclose(self, other)
../../../micromamba/envs/gha-test-env/lib/python3.10/site-packages/numpy/_core/numeric.py:2329: in allclose
    res = all(isclose(a, b, rtol=rtol, atol=atol, equal_nan=equal_nan))
E   RecursionError: maximum recursion depth exceeded while calling a Python object
!!! Recursion detected (same locals & position)
```


It's not clear why they started failing now. Here's a diff of installed dependencies between failing and successfully jobs

<details><summary>Installed deps</summary>
<p>

```diff
< _libgcc_mutex              0.1           
18c17
< coverage                   7.13.2        
---
> coverage                   7.13.4        
28,29c27,28
< filelock                   3.20.3        
< flask                      3.1.2         
---
> filelock                   3.24.3        
> flask                      3.1.3         
34c33
< fontconfig                 2.15.0        
---
> fontconfig                 2.17.1        
42c41
< greenlet                   3.3.1         
---
> greenlet                   3.3.2         
57c56
< ld_impl_linux-64           2.45          
---
> ld_impl_linux-64           2.45.1        
69c68
< libexpat                   2.7.3         
---
> libexpat                   2.7.4         
78c77
< libglib                    2.86.3        
---
> libglib                    2.86.4        
87,89c86,88
< libpng                     1.6.54        
< libpq                      18.1          
< librdkit                   2025.09.4     
---
> libpng                     1.6.55        
> libpq                      18.2          
> librdkit                   2025.09.5     
116d114
< patsy                      1.0.2         
118,119c116,117
< pillow                     12.1.0        
< pip                        25.3          
---
> pillow                     12.1.1        
> pip                        26.0.1        
121c119
< platformdirs               4.5.1         
---
> platformdirs               4.9.2         
127,128c125,126
< pybind11                   3.0.1         
< pybind11-global            3.0.1         
---
> pybind11                   3.0.2         
> pybind11-global            3.0.2         
142,143c140,141
< quippy-ase                 0.10.1        
< rdkit                      2025.09.4     
---
> quippy-ase                 0.10.2        
> rdkit                      2025.09.5     
145c143
< reportlab                  4.4.9         
---
> reportlab                  4.4.10        
149,151c147
< setuptools                 80.10.2       
---
> setuptools                 82.0.0        
156d151
< statsmodels                0.14.6        
167,169c162,164
< unicodedata2               17.0.0
< virtualenv                 20.36.1
< werkzeug                   3.1.5   
---
> unicodedata2               17.0.1        
> virtualenv                 20.38.0       
> werkzeug                   3.1.6         
173c168
< xorg-libx11                1.8.12  
---
> xorg-libx11                1.8.13        
181c176
< zlib-ng                    2.3.2   
---
> zlib-ng                    2.3.3
```

</p>
</details> 

#### ChatGPTs analysis (likely wrong)

```
The job failed due to RecursionError: maximum recursion depth exceeded in comparison in the following tests:

- tests/test_configuration.py::test_solvate
- tests/test_configuration_set.py::test_configurations_load_xyz_with_energies_forces

Both errors involve the solvate method and comparison operations in the Configuration class.

**Root Cause**  
The RecursionError likely comes from the __eq__ method in mlptrain/configurations/configuration.py, which compares self.box == other.box. If Box's __eq__ method is not properly implemented (e.g., if it calls Configuration's __eq__ in a loop or doesn't terminate), recursion depth is exceeded.

**Solution Steps**  

1. Check Box class's __eq__ implementation. Ensure it only compares Box properties and does not reference Configuration objects.

2. In the Configuration class's __eq__ method, add a safeguard to ensure self.box and other.box are not referencing each other recursively:

Replace:
```python
577|             and other.box == self.box
```
with:
```python
577|             and (self.box is other.box or self.box == other.box)
```

If Box is None, handle explicitly:
```python
575|             and ((self.box is None and other.box is None) or (self.box == other.box))
```

3. Double-check the solvate method’s creation and assignment of self.box = Box([box_size] * 3) to ensure it doesn’t introduce unintended recursion via attributes.

**Example Code Fix for Configuration.__eq__**  
Update __eq__ to:
```python
570|     def __eq__(self, other) -> bool:
571|         """Another configuration is identical to this one"""
572|         eq = (
573|             isinstance(other, Configuration)
574|             and other.n_atoms == self.n_atoms
575|             and other.mult == self.mult
576|             and other.charge == self.charge
577|             and ((self.box is None and other.box is None) or (self.box == other.box))
578|         )
579|         if eq and self.n_atoms > 0:
580|             rmsd = np.linalg.norm(self.coordinates - other.coordinates)
581|             return eq and rmsd < 1e-10
582|         return eq
```
**DeprecationWarnings**  
You may also want to update usages of np.array(cell, float)/np.array(cell) in configuration.py (lines 694) due to DeprecationWarning. These need to add copy=True/False if required.

**Summary**  
Fix RecursionError by updating the __eq__ method to handle box comparison without triggering recursion. Review Box's equality implementation and adjust as above. This will resolve the test failures and errors in your CI job.
```

--- a/environment.yml
+++ b/environment.yml
@@ -9,7 +9,7 @@ dependencies:
   - pip
   - pre-commit
   - ase
-  - autode=1.3.3
+  - autode=1.4.5
   - coloredlogs
   - cython
   - dscribe=2.0

--- a/environment_ace.yml
+++ b/environment_ace.yml
@@ -9,7 +9,7 @@ dependencies:
   - pip
   - pre-commit
   - ase
-  - autode=1.3.3
+  - autode=1.4.5
   - coloredlogs
   - cython
   - dscribe=2.0

--- a/environment_mace.yml
+++ b/environment_mace.yml
@@ -10,7 +10,7 @@ dependencies:
   - libgfortran=14.*
   - pre-commit
   - ase
-  - autode=1.3.3
+  - autode=1.4.5
   - coloredlogs
   - cython
   - dscribe=2.0

